iothub/device/src/Transport/Http/HttpClientHelper.cs

```diff
@@ -55,9 +55,24 @@ internal sealed class HttpClientHelper : IDisposable
             _baseAddress = baseAddress;
             _connectionCredentials = connectionCredentials;
             _defaultErrorMapping = new ReadOnlyDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>>(defaultErrorMapping);
-            _httpClientHandler = httpClientHandler ?? new HttpClientHandler();
-            _httpClientHandler.SslProtocols = iotHubClientHttpSettings.SslProtocols;
-            _httpClientHandler.CheckCertificateRevocationList = iotHubClientHttpSettings.CertificateRevocationCheck;
+
+#if NETCOREAPP
```

Comment:
Even though we no longer target .NET core versions, my understanding is that this block will still execute if a user with a .NET core app takes a dependency on our SDK. Please correct me if I'm wrong on this one, though

Reply:
A preprocessor directive exists only in sourcecode = pre-compiled code. When we compile our code and build a dll, the #if-defs should already have been compiled based on the condition and the targets defined in the project.
So, if I understand this correctly, our v2 dll (which is ns2.0 only) will not have this if-def code compiled, so this will essentially be unreachable code.

You can do a local build (dll) + pack (nuget) and confirm this.

Reply:
Interesting. I'll need to rethink this strategy a bit then.